### ISSUE TITLE
headerConfig.prefix should default to empty

### ISSUE THREAD

**Issue by noahjahn** [hydra-newmedia/passport-headerapikey#8]
Not setting the `headerConfig.prefix` value results in the middleware function never being called.

For example, here is where I'm using the strategy:

```
passport.use(new HeaderAPIKeyStrategy(
    { header: 'X-Api-Key' },
    false,
    function (apikey, done) {
        console.log('hello from headerapikeystrategy');
        User.findOne({ apikey: apikey }, (err, user) => {
            if (err) {
                console.error(err);
                return done(err);
            }
            if (!user) {
                return done(null, false);
            }
            return done(null, user);
        });
    }
));
```

And using this strategy as express middleware: 

```
router.post('/login', passport.authenticate('headerapikey', { session: false }), (req, res) => {
    console.log('authenticated!');
    res.status(200).sendFormat(req, res);
});
```

The server always responds with 401 Unauthorized and I never see 'hello from headerapikeystrategy' printed out in the console.

---

To fix this, I had to pass an empty string to the `headerConfig.prefix` value in the constructor:  

```
passport.use(new HeaderAPIKeyStrategy(
    { header: 'X-Api-Key', prefix: '' },
    false,
    function (apikey, done) {
        console.log('hello from headerapikeystrategy');
        User.findOne({ apikey: apikey }, (err, user) => {
            if (err) {
                console.error(err);
                return done(err);
            }
            if (!user) {
                return done(null, false);
            }
            return done(null, user);
        });
    }
));
```

Not setting the prefix should default to empty.

**Comment by guischdi** [comment]
Hi @noahjahn 
Thanks a lot for your issue. Trying to fix this tomorrow.

**Comment by guischdi** [comment]
published as [v1.2.2](https://www.npmjs.com/package/passport-headerapikey/v/1.2.2)
